### PR TITLE
Finished payment model, started tests

### PR DESCRIPTION
**Pull request by odoric27**
havne't done much with the tests yet, but there's a simple test_payments.py here now for people to get started with.

currently still attached to sqlite, but i will update that once postgres requirements merged into master.

i think this model makes the most sense with what i've experimented with thus far. Give us just one main Payment objects with attached Detail objects to hold specific info w/r/t credit vs debit vs paypal

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+        except KeyError as e:
+            raise DataValidationError('Invalid payment: missing ' + e.args[0])
+        except TypeError as e:
+            raise DataValidationError('Invalid payment: body of request contained bad or no data')
+        return self
+
+
+'''
+class CreditPayment(db.Model):
+    __tablename__ = 'credit-cards'
```
Comment (odoric27) [inline models.py]: yeah, i'll delete before the merge.

File: models.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+            d = Detail()
+            if (self.payment_type == 'credit' or self.payment_type == 'debit'):
+                self.details = d.deserialize_card(details)
+            else:
+                self.details = d.deserialize_paypal(details)
+        except KeyError as e:
+            raise DataValidationError('Invalid payment: missing ' + e.args[0])
+        except TypeError as e:
+            raise DataValidationError('Invalid payment: body of request contained bad or no data')
+        return self
```
Comment (odoric27) [inline models.py]: I.... don't know. An artifact from starting with the lecture materials, which do the same. But looking back at the TDD lecture code, i don't know why Rofrano's deserialize returns self either. Either way, striking the returns.

File: payments.py [hunk 3]
```diff
@@ -5,11 +5,17 @@
...
 import time
 from flask import Flask, jsonify, request, make_response, Response, json, url_for
+from flask_sqlalchemy import SQLAlchemy
 
 # Create Flask application
 app = Flask(__name__)
 #app.config['LOGGING_LEVEL'] = logging.INFO
 
+# DB Config
+app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///data/dev.db'
```
Comment (odoric27) [inline payments.py]: good call. thanks

File: payments.py [hunk 4]
```diff
@@ -5,11 +5,17 @@
...
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+db = SQLAlchemy(app)
+
```
Comment (odoric27) [inline payments.py]: yeah, makes sense. i'll add this to your existing module __init__ once merged in

File: test_payments.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+        payment = models.Payment()
+        payment.deserialize(data)
+        payments.db.session.add(payment)
+        payments.db.session.commit()
+        self.app = payments.app.test_client()
+
+    def tearDown(self):
+        payments.db.session.remove()
+        payments.db.drop_all()
```
Comment (odoric27) [inline test_payments.py]: i'll have to look into this one myself. the tearDown was lifted from lecture code just to get started

**Comment by odoric27** [comment]
@jdgumz changes are up if/when you're able to take a look.

didn't implement the pulling of the connection string from environment yet, but can do so later. still need to read up on that tearDown method as well, but for now, it seems to work. will report back.

also - using this obviously relies on items in #50 so if anyone @SharanyaT @nvme5 (is there a way to just @group?) can please take a look and merge that would be much appreciated!

File: Vagrantfile [hunk 6]
```diff
@@ -44,17 +39,51 @@ Vagrant.configure(2) do |config|
...
     rm cf-cli-installer_6.24.0_x86-64.deb
     # Install app dependencies
     cd /vagrant
     sudo pip install -r requirements.txt
     # Make vi look nice
     echo "colorscheme desert" > ~/.vimrc
+
+    cat <<-EOF | su - postgres -c psql
+    -- Create the database user:
+    CREATE USER payments WITH PASSWORD 'payments';
```
Comment (jdgumz) [inline Vagrantfile]: Is it possible to not hardcode the password value?  Although I do realize it's for non-prod data.

Comment (odoric27) [inline Vagrantfile]: i imagine there is, but my rationale for this down and dirty approach was precisely as you say: "it's not prod data."

something i can look into for our local environs, but certainly something we'll implement when we get the service running on bluemix

**Pull request merged**